File: authsphere-admin/src/api/passwordPolicy.ts
import http from './http'

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export interface PasswordPolicyQuery {
  page: number
  size: number
  code?: string
  name?: string
  status?: number
}

export interface PasswordPolicyPayload {
  code: string
  name: string
  minLength: number
  maxLength: number
  requireUppercase: boolean
  requireLowercase: boolean
  requireDigit: boolean
  requireSpecialChar: boolean
  disallowUsername: boolean
  historyCount: number
  expireDays: number
  retryLimit: number
  lockMinutes: number
  description?: string
}

export interface PasswordPolicyRecord extends PasswordPolicyPayload {
  id: string
  status: number
  createTime?: string
  updateTime?: string
}

export interface PasswordPolicyListItem {
  id: string
  code: string
  name: string
  minLength: number
  maxLength: number
  expireDays: number
  retryLimit: number
  lockMinutes: number
  status: number
  createTime?: string
  updateTime?: string
}

export const passwordPolicyApi = {
  page(params: PasswordPolicyQuery) {
    return http.post<unknown, PageResult<PasswordPolicyRecord>>('/admin/password/policy/page', params)
  },
  detail(id: string) {
    return http.get<unknown, PasswordPolicyRecord>(`/admin/password/policy/${id}`)
  },
  create(payload: PasswordPolicyPayload) {
    return http.post<unknown, PasswordPolicyRecord>('/admin/password/policy', payload)
  },
  update(id: string, payload: PasswordPolicyPayload) {
    return http.put<unknown, PasswordPolicyRecord>(`/admin/password/policy/${id}`, payload)
  },
  toggleStatus(id: string) {
    return http.put<unknown, boolean>(`/admin/password/policy/status/${id}`)
  },
  list() {
    return http.get<unknown, PasswordPolicyListItem[]>('/admin/password/policy/list')
  },
}

File: authsphere-server/authsphere-module-realm/src/main/java/com/authsphere/server/realm/model/PasswordPolicy.java
package com.authsphere.server.realm.model;

import com.authsphere.server.common.model.BaseDataBaseModel;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 身份域密码策略。
 */
@Data
@TableName(value = "password_policy")
@EqualsAndHashCode(callSuper = true)
public class PasswordPolicy extends BaseDataBaseModel {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 策略编码。
     */
    private String code;

    /**
     * 策略名称。
     */
    private String name;

    /**
     * 最小密码长度。
     */
    private Integer minLength;

    /**
     * 最大密码长度。
     */
    private Integer maxLength;

    /**
     * 是否要求大写字母。
     */
    private Boolean requireUppercase;

    /**
     * 是否要求小写字母。
     */
    private Boolean requireLowercase;

    /**
     * 是否要求数字。
     */
    private Boolean requireDigit;

    /**
     * 是否要求特殊字符。
     */
    private Boolean requireSpecialChar;

    /**
     * 是否禁止密码包含用户名。
     */
    private Boolean disallowUsername;

    /**
     * 历史密码不可重复次数。
     */
    private Integer historyCount;

    /**
     * 密码有效期天数，0 表示不限制。
     */
    private Integer expireDays;

    /**
     * 连续失败锁定阈值，0 表示不启用。
     */
    private Integer retryLimit;

    /**
     * 锁定分钟数，0 表示不自动解锁。
     */
    private Integer lockMinutes;

    /**
     * 状态。
     */
    private Integer status;

    /**
     * 描述。
     */
    private String description;
}
